### ISSUE TITLE
accel_perf example cannot be gracefully terminated when --wait-for-rpc is used

### ISSUE THREAD

**Issue by karlatec** [spdk/spdk#3160]
# Sighting report

As in title - accel_perf example cannot be gracefully terminated when --wait-for-rpc option is used.
The only way to shut it down is with SIGKILL.

## Expected Behavior

Possible to gracefully shutdown `accel_perf` binary.

## Current Behavior

Does not work as expected.

## Steps to Reproduce

1. Run `accel_perf`:
```
[sys_sgci@spdk-ac-02 spdk]$ sudo build/examples/accel_perf -w copy --wait-for-rpc -y
[2023-10-20 19:56:20.732841] Starting SPDK v24.01-pre git sha1 631e75fc0 / DPDK 23.07.0 initialization...
[2023-10-20 19:56:20.732999] [ DPDK EAL parameters: accel_perf --no-shconf -c 0x1 --huge-unlink --no-telemetry --log-level=lib.eal:6 --log-level=lib.cryptodev:5 --log-level=user1:6 --base-virtaddr=0x200000000000 --match-allocations --file-prefix=spdk_pid804950 ]
EAL: No free 2048 kB hugepages reported on node 1
[2023-10-20 19:56:20.840584] app.c: 794:spdk_app_start: *NOTICE*: Total cores available: 1
[2023-10-20 19:56:20.943062] reactor.c: 937:reactor_run: *NOTICE*: Reactor started on core 0
```
2. Press CTRL-C a couple of times. Process still present.
```
[2023-10-20 19:56:20.840584] app.c: 794:spdk_app_start: *NOTICE*: Total cores available: 1
[2023-10-20 19:56:20.943062] reactor.c: 937:reactor_run: *NOTICE*: Reactor started on core 0
bz^Bz^C^C^C^C^C
^C^C^C^C

[sys_sgci@spdk-ac-02 ~]$ pgrep reactor
804950
```
3. Try to kill with SIGTERM and SIGINT. Process still present.
```
[sys_sgci@spdk-ac-02 ~]$ sudo kill -2 $(pgrep reactor); pgrep reactor
804950
[sys_sgci@spdk-ac-02 ~]$ sudo kill -15 $(pgrep reactor); pgrep reactor
804950
```

4. Finally - kill with SIGKILL
```
[sys_sgci@spdk-ac-02 ~]$ sudo kill -9 $(pgrep reactor); pgrep reactor
[sys_sgci@spdk-ac-02 ~]$ pgrep reactor
[sys_sgci@spdk-ac-02 ~]$
```

## Context (Environment including OS version, SPDK version, etc.)

Linux spdk-ac-02 6.1.14-200.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Sun Feb 26 00:13:26 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
gcc (GCC) 12.2.1 20221121 (Red Hat 12.2.1-4)

SPDK 631e75fc03a79bb5fe45a53d293e30bfabd947d8

**Comment by ksztyber** [comment]
I think this should already be fixed by da5ec81f429c21c73039c3c3bd3515be7670715a.

**Comment by karlatec** [comment]
https://github.com/spdk/spdk/commit/da5ec81f429c21c73039c3c3bd3515be7670715a was merged.
I retested reported steps with fix built and issue is no longer reproducible. Closing.
